A generated Kotlin project with Gradle doesn't build

**Describe the bug**
When generating a project like so:

```
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create     -DprojectGroupId=org.acme     -DprojectArtifactId=gradle-kotlin-test     -DclassName="org.acme.GreetingResource"     -Dpath="/greeting"     -Dextensions="kotlin" -DplatformArtifactId=quarkus-bom -DbuildTool=gradle
```

and then trying to build it like so:

```
gradle build -x test
```

I get:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/gandrian/Desktop/gradle-kotlin-test/build.gradle' line: 27

* What went wrong:
A problem occurred evaluating root project 'gradle-kotlin-test'.
> Could not find method setOutputDirectory() for arguments [/home/gandrian/Desktop/gradle-kotlin-test/build/classes/kotlin/main] on object of type io.quarkus.gradle.QuarkusPluginExtension.

```

**Environment (please complete the following information):**
 - Quarkus version or git rev: latest `master`

--- a/devtools/gradle/pom.xml
+++ b/devtools/gradle/pom.xml
@@ -64,6 +64,7 @@
                                 <argument>${gradle.task}</argument>
                                 <argument>-Pdescription=${project.description}</argument>
                                 <argument>-S</argument>
+                                <argument>--stacktrace</argument>
                             </arguments>
                             <environmentVariables>
                                 <MAVEN_LOCAL_REPO>${settings.localRepository}</MAVEN_LOCAL_REPO>

--- a/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
+++ b/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
@@ -46,7 +46,8 @@ public class QuarkusPluginFunctionalTest {
     }
 
     @ParameterizedTest(name = "Build {0} project")
-    @EnumSource(SourceType.class)
+    //TODO: Fix Scala build in Windows
+    @EnumSource(value = SourceType.class, names = {"JAVA","KOTLIN"})
     public void canBuild(SourceType sourceType) throws IOException {
         createProject(sourceType);
 

--- a/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
+++ b/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
@@ -9,19 +9,30 @@ import java.util.List;
 import io.quarkus.cli.commands.CreateProject;
 import io.quarkus.cli.commands.writer.FileProjectWriter;
 import io.quarkus.generators.BuildTool;
+import io.quarkus.generators.SourceType;
 import org.gradle.testkit.runner.BuildResult;
 import org.gradle.testkit.runner.GradleRunner;
 import org.gradle.testkit.runner.TaskOutcome;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class QuarkusPluginFunctionalTest {
 
+    private File projectRoot;
+
+    @BeforeEach
+    void setUp(@TempDir File projectRoot) {
+        this.projectRoot = projectRoot;
+    }
+
     @Test
-    public void canRunListExtensions(@TempDir File projectRoot) throws IOException {
-        createProject(projectRoot);
+    public void canRunListExtensions() throws IOException {
+        createProject(SourceType.JAVA);
 
         BuildResult build = GradleRunner.create()
                 .forwardOutput()
@@ -34,9 +45,10 @@ public class QuarkusPluginFunctionalTest {
         assertThat(build.getOutput()).contains("Quarkus - Core");
     }
 
-    @Test
-    public void canBuild(@TempDir File projectRoot) throws IOException {
-        createProject(projectRoot);
+    @ParameterizedTest(name = "Build {0} project")
+    @EnumSource(SourceType.class)
+    public void canBuild(SourceType sourceType) throws IOException {
+        createProject(sourceType);
 
         BuildResult build = GradleRunner.create()
                 .forwardOutput()
@@ -60,12 +72,13 @@ public class QuarkusPluginFunctionalTest {
         return arguments;
     }
 
-    private void createProject(@TempDir File projectRoot) throws IOException {
+    private void createProject(SourceType sourceType) throws IOException {
         assertThat(new CreateProject(new FileProjectWriter(projectRoot))
                            .groupId("com.acme.foo")
                            .artifactId("foo")
                            .version("1.0.0-SNAPSHOT")
                            .buildTool(BuildTool.GRADLE)
+                           .sourceType(sourceType)
                            .doCreateProject(new HashMap<>()))
                 .withFailMessage("Project was not created")
                 .isTrue();

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPluginExtension.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPluginExtension.java
@@ -36,12 +36,20 @@ public class QuarkusPluginExtension {
         return new File(outputDirectory);
     }
 
+    public void setOutputDirectory(String outputDirectory) {
+        this.outputDirectory = outputDirectory;
+    }
+
     public File outputConfigDirectory() {
         if (outputConfigDirectory == null) {
             outputConfigDirectory = project.getConvention().getPlugin(JavaPluginConvention.class)
                     .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput().getResourcesDir().getAbsolutePath();
         }
         return new File(outputConfigDirectory);
+    }
+
+    public void setOutputConfigDirectory(String outputConfigDirectory) {
+        this.outputConfigDirectory = outputConfigDirectory;
     }
 
     public File sourceDir() {
@@ -52,6 +60,10 @@ public class QuarkusPluginExtension {
         return new File(sourceDir);
     }
 
+    public void setSourceDir(String sourceDir) {
+        this.sourceDir = sourceDir;
+    }
+
     public File workingDir() {
         if (workingDir == null) {
             workingDir = outputDirectory().getPath();
@@ -60,11 +72,19 @@ public class QuarkusPluginExtension {
         return new File(workingDir);
     }
 
+    public void setWorkingDir(String workingDir) {
+        this.workingDir = workingDir;
+    }
+
     public String finalName() {
         if (finalName == null || finalName.length() == 0) {
             this.finalName = String.format("%s-%s", project.getName(), project.getVersion());
         }
         return finalName;
+    }
+
+    public void setFinalName(String finalName) {
+        this.finalName = finalName;
     }
 
     public Set<File> resourcesDir() {

